File: frontend/src/types/llm_config.ts
import type { Provider } from './common'

export interface LLMConfig {
  id: number
  name: string
  provider: Provider
  api_key: string // server returns masked
  api_base: string
  model: string
  temperature: number
  max_tokens: number
  extra: Record<string, unknown>
  is_active: boolean
}

export interface LLMConfigCreate {
  name: string
  provider: Provider
  api_key: string
  api_base?: string
  model: string
  temperature?: number
  max_tokens?: number
  extra?: Record<string, unknown>
}

export type LLMConfigUpdate = Partial<LLMConfigCreate>

export interface LLMTestResult {
  ok: boolean
  message: string
  latency_ms: number
}

File: app/models/llm_config.py
"""LLMConfig ORM model."""
from sqlalchemy import Boolean, Float, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin


class LLMConfig(Base, TimestampMixin):
    __tablename__ = "llm_configs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(100), unique=True, nullable=False)
    provider: Mapped[str] = mapped_column(String(50), nullable=False)
    api_key: Mapped[str] = mapped_column(Text, nullable=False)
    api_base: Mapped[str] = mapped_column(String(500), nullable=False, default="")
    model: Mapped[str] = mapped_column(String(100), nullable=False)
    temperature: Mapped[float] = mapped_column(Float, nullable=False, default=0.7)
    max_tokens: Mapped[int] = mapped_column(Integer, nullable=False, default=2048)
    extra: Mapped[str] = mapped_column(Text, nullable=False, default="{}")  # JSON
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)

    def __repr__(self) -> str:
        return f"<LLMConfig {self.name} provider={self.provider} model={self.model}>"
